app/experimenter/experiments/forms.py

```diff
@@ -434,19 +430,47 @@ def is_valid(self):
         return super().is_valid() and self.variants_formset.is_valid()
 
     def save(self, *args, **kwargs):
-        experiment = super().save(*args, **kwargs)
+        self.variants_formset.save()
```

Comment:
Two saves should be done within the same transaction. You can take a big bite out of https://github.com/mozilla/experimenter/issues/1062 by doing a...:

```python
with transaction.atomic():
    self.variants_formset.save()
    return super().save(*args, **kwargs)
```

It is admittedly a bit off-topic but that's how it's supposed by be done anyway and it'd be a shame to not tackle it on edited code.